scripts/user_data_base.ps1

```diff
@@ -0,0 +1,163 @@
+             'sa-east-1'= 'sae1';
+             'us-east-1'= 'use1';
+             'us-west-1'= 'usw1';
+             'us-west-2'= 'usw2'}
+
+
+Set-DefaultAWSRegion -Region $region
+try {
+  New-Item $BASE_DIR -type directory
+  New-Item $CONFIG -type directory
+  New-Item $LOG_DIR -type directory
+} catch {}
+
+
+function New-ServerName {
+    $serverId=GetNewId
+    $zone = Get-MyAvailabilityZone
+    $region = Get-MyRegion
+    $zoneLetter = $zone -replace $region, ""
+    $role = Get-MyRole
+    $regionName = $REGIONS[$region]
+    $myName="$role-$($regionName)$zoneLetter-$serverId"
+    return $myName
+}
+
+function GetNewId
+{
+    $ticks = [System.DateTime]::Now.Ticks;
+    $ticks = $ticks -shr 16;
+    $ticks = $ticks -band 0x3fffffff;
+    return ZBase32Encode($ticks);
+}
+
+try{
+    Set-Variable ZBase32Alphabet -Option ReadOnly -Scope script -Value "ybndrfg8ejkmcpqxot1uwisza345h769"
+    Set-Variable ZBase32Base -Option ReadOnly -Scope script -Value 32
+}catch {}
+function ZBase32Encode([long] $number)
+{
+    if ($number -eq 0)
+    {
+        return $ZBase32Alphabet[0].ToString();
+    }
+    $encoded = New-Object System.Text.StringBuilder(64);
+    while ($number -gt 0)
+    {
+        $character = $ZBase32Alphabet[[int]($number % $ZBase32Base)];
+        $ignore = $encoded.Insert(0, $character);
+        $number = [int][System.Math]::Floor($number / $ZBase32Base);
+    }
+    
+    return $encoded.ToString();
+}
+
+function Get-MyAvailabilityZone {
+   
+    return (Invoke-WebRequest -UseBasicParsing -Uri $METADATA_BASEURL/placement/availability-zone).Content
+    
+}
+
+function Get-MyRegion {
+    return ((Invoke-WebRequest -UseBasicParsing  -Uri $DYNAMICDATA_BASEURL/instance-identity/document ).Content | ConvertFrom-Json).region
+}
+
+function Get-MyRole {
+    try{
+    $result = Invoke-WebRequest -UseBasicParsing -Uri $METADATA_BASEURL/iam/security-credentials/
+    $role = $result.Content
+    return $role
+    } catch {
+        return ""
+    }
+}
+
+
+### Determine server name and set it
+### Generate client.rb
+### Install chef
+$servername = New-ServerName
+
+# Store and Set servername in config dir
+Write-Output $servername | Out-File -FilePath ($CONFIG + "\servername.txt") -ErrorAction Stop
+$File = Get-Content "c:\hudl\config\servername.txt"
+[System.IO.File]::WriteAllLines("c:\hudl\config\servername_chef", $File)
+New-EC2Tag -Region $region -ResourceId $instanceId -Tag @{key="Name"; value="$servername"}
+
+# Install Chef
+$chef_tag_role = Get-EC2Tag -Filters @( @{Name="key"; values="Role"}, @{Name="resource-id"; Values=$instanceId})
+$chef_environment = Get-EC2Tag -Filters @( @{Name="key"; values="Environment"}, @{Name="resource-id"; Values=$instanceId})
+$windows_version = "2012r2"
+$chef_version = "chef-client-12.6.0-1-x86.msi"
+$server_name = $servername.Trim()
+$chef_env = $chef_environment.Value.ToLower().Trim()
+$client_rb = @"
+base_dir = "C:/Chef"
+log_level        :info
+log_location     STDOUT
```

Comment:
@jamiegs We could set it there - the chef-client cookbook automatically creates a default log location and log file in C:\chef\log